uber/models.py

```diff
@@ -1243,9 +1196,10 @@ def _badge_adjustments(self):
 
         self.badge_type = get_real_badge_type(self.badge_type)
 
+        if self.orig_value_of('badge_type') != self.badge_type:
+            self.session.update_badge(self, self.orig_value_of('badge_type'), self.orig_value_of('badge_num'))
```

Comment:
Uhhhhhhhhhh.

Hm. I did a lot of going in circles when working on this and I think I might have ended up a bit turned around. I THINK the intent was to do this instead of calling update_badge directly in the admin form, but I never removed the update_badge call.

I'll have to load this feature branch back into local and test it again.